src/core/qgsofflineediting.cpp

```diff
@@ -1492,7 +1492,15 @@ void QgsOfflineEditing::committedFeaturesAdded( const QString &qgisLayerId, cons
   }
   else
   {
-    tableName = uri.param( offlinePath + "|layername" );
+    if ( dataSourceString.indexOf( QLatin1String( "|layername=" ) ) != -1 )
```

Comment:
Why not use QgsDataProviderRegistry::decodeUri instead?

Reply:
I was looking for something like `layer->dataProvider().staticMetaData.decodeUri()` but didn't find any.

Reply:
Nope, you do it through the registry using the layer's provider key

Reply:
Ok. If this is the preferred approach I can do that.

Reply:
Yep, it's definitely preferred. If nothing else it's covered by unit tests...

Reply:
Done :+1: